department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-33: Queue Column Rework

Resolves #{github issue number}

### Description
This PR consists of two distinct yet related parts. There was some overlap due to how tightly coupled queue/tab configuration is with backend task retrieval and sorting.

1. Added a generic method for adding a default sort to a queue tab. Going forward this will allow an organization that utilizes QueueTableBuilder to choose a default sorting column and direction by simply adding a method that contains a column to initially sort on if a sort by direction is not already chosen by the get parameters.

2.Improving the receipt_date column sort query. The receipt date query was not very efficient and I think this implementation (if identical) is much quicker since it stays more in the SQL realm instead of the Ruby realm.  There's also a small refactor to task_sorter to work with this new sort and queue_config to automatically sort using the correct column instead of a specific sort case for the receipt_date column inside of task_sorter.

### Acceptance Criteria
- [ ] No regressions in other QueueTable apps
- [ ] No additional failing tests
- [ ] The query is an improvement in load time

### Testing Plan
1. Log in as a BVA intake user BVAISHAW will work
2. Visit the BVA intake queue using the url relative /organizations/bva-intake
3. Verify the get parameters are correct with the receipt date column and ascending search order
![image](https://user-images.githubusercontent.com/109369527/208758922-43098e4c-6d32-4f63-b116-493e331293c1.png)

4. Verify that the column is sorted correctly in the UI and data
![image](https://user-images.githubusercontent.com/109369527/208759046-425977bc-6a61-4b74-88cf-37894377bde9.png)

5. Verify that sorting works and the get parameters are changed to desc for the receipt date column
![image](https://user-images.githubusercontent.com/109369527/208759563-3fe944e4-e0e1-4688-81e6-56fdae0a8da8.png)

6. Verify that other sorting is not adversely affected by these changes
![image](https://user-images.githubusercontent.com/109369527/208759685-3c7cb07c-bae5-42d6-b6ed-fd00b0fee004.png)

7. Verify that the Appeal type is properly displayed in the UI for a different queue now
8. Test that the query is quicker either with an eye test through loading both versions or benchmarking
